src/opera/test/pge/test_metfile.py

```diff
@@ -0,0 +1,214 @@
+# export licenses, or other export authority as may be required, before
+# exporting such information to foreign countries or providing access to
+# foreign persons.
+#
+
+"""
+=================
+test_metfile.py
+=================
+
+Unit tests for the util/metfile.py module.
+"""
+import fileinput
+import os
+import shutil
+import tempfile
+import unittest
+
+from os.path import abspath, join
+
+from pkg_resources import resource_filename
+
+from opera.pge import PgeExecutor, RunConfig
+from opera.util.metfile import MetFile
+from opera.util import PgeLogger
+
+
+class MetFileTestCase(unittest.TestCase):
+    """Base test class using unittest"""
+
+    TEST_MET_FILE = resource_filename('opera', '/test/pge/base_pge_test/scratch/testMetFile.met')
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+        Set up directories for testing
+        Initialize regular expression
+        Initialize other class variables
+
+        """
+        cls.starting_dir = abspath(os.curdir)
+        cls.test_dir = resource_filename(__name__, "")
+        cls.data_dir = join(cls.test_dir, "data")
+
+        os.chdir(cls.test_dir)
+
+        cls.working_dir = tempfile.TemporaryDirectory(
+            prefix="test_time_", suffix='temp', dir=os.curdir)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        """
+        At completion re-establish starting directory
+        -------
+        """
+        cls.working_dir.cleanup()
+        os.chdir(cls.starting_dir)
+
+    def setUp(self) -> None:
+        """
+        Use the temporary directory as the working directory
+        -------
+        """
+        os.chdir(self.working_dir.name)
+
+    def tearDown(self) -> None:
+        """
+        Return to starting directory
+        -------
+        """
+        os.chdir(self.test_dir)
+
+    def testMetFile(self):
+        """
+        Tests instantiation of MetFile class
+        Methods tested in MetFile class:
+            set_key_value()
+            write_met_file()
+            read_met_file()
+
+        """
+        met_file = MetFileTestCase.TEST_MET_FILE
+        met_data = MetFile(met_file)
+        self.assertIsInstance(met_data, MetFile)
+
+        # Test __setitem__ and __getitem__
+        met_data["test key"] = "test value"
+        self.assertEqual(met_data["test key"], "test value")
+        # Test write_met_file()
+        met_data.write_met_file()
+        # Copy contents of the file into a variable
+        with open(met_file, 'r') as mf:
+            lines = (mf.readlines())
+        # Test string to look for string in the dictionary
+        test_str = '  "test key": "test value",\n'
+        self.assertIn(test_str, lines)
+        # Test read_met_file()
+        met_dict = met_data.read_met_file()
+        self.assertEqual(met_data['test key'], 'test value')
+
+        # Read another line to test for an existing file
+        met_dict = "test key 2", "test value 2"
```

Comment:
Not sure what this line is doing or is still needed? `met_dict` gets assigned again on line 121 before its used with this tuple (?)

Reply:
This whole section was redone. The simple test met file is generated and tested during the test and is deleted when the TemporaryDirectory goes away.